src/Cli/dotnet/Commands/Run/VirtualProjectBuildingCommand.cs

```diff
@@ -192,14 +197,28 @@ ProjectRootElement CreateProjectRootElement(ProjectCollection projectCollection)
             projectRoot.FullPath = projectFileFullPath;
             return projectRoot;
         }
+
+        static string GetArtifactsPath(string entryPointFilePath)
+        {
+            // We want a location where permissions are expected to be restricted to the current user.
+            var directory = RuntimeInformation.IsOSPlatform(OSPlatform.Windows)
+                ? Path.GetTempPath()
+                : Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData);
+            return Path.Join(directory, "dotnet", "runfile", Sha256Hasher.HashWithNormalizedCasing(entryPointFilePath));
```

Comment:
It could be good to prefix the hash with the file name (e.g. `app` in `/path/to/app.cs`) in case we ever need to go in here and understand what things mean.